docs/FinalReport.md

```diff
@@ -0,0 +1,86 @@
+
+## Feature Description and Requirements
+
+1. Users can register, login and rate movies they've seen.
+2. Users can link up with their friends and generate a "Top 20" list of movies that they will most likely enjoy together. 
+3. Users can change their name and password. 
+4. Users are able to add and delete friends from their friends list, and from their friends list create groups.
+
+## Design
+
+![Design](screenshots/design.png)
+
+![Updated Class Diagram](screenshots/classDiagram1.png)
+
+![Updated Class Diagram](screenshots/classDiagram2.png)
+
+## User Interface
+
+![Login](screenshots/login.png)
+
+![Register](screenshots/register.png)
+
+![Movie Search w/ Rating](screenshots/movie_search_wrating.png)
+
+![Friends List](screenshots/friendslist.png)
+
+![Find Friends](screenshots/findfriends.png)
+
+![Add Friends](screenshots/addfriends.png)
+
+![Make Group](screenshots/makegroup.png)
+
+![Add Group](screenshots/addedgroup.png)
+
+![Add friends to group](screenshots/addfriendstogroup.png)
+
+![Group has members](screenshots/grouphasfriends.png)
+
+![Group Recommendations](screenshots/grouprecommendations.png)
+
+![Settings](screenshots/settings.png)
+
+![Change Name](screenshots/changename.png)
+
+## Testing Tools, Test Scenarios, and Test Cases
+
+Our application was divided into two different parts for testing, the first is the API (back end) and the second is the UI (front end). The team’s plan was to make sure that with each pull request on git hub our systems would be tested. 
+For each class within the API a JUnit test was created, this way we could test the functionality of each class separately. The Rating class was tested to make sure that a user could easily rate any movie, the loading, saving, and modifying aspects were also tested. The User class was tested so that each User could add friends, delete friends, create groups, delete groups, and login. The Session class was mainly tested to make sure that a session would be deleted from our database after x amount of time. The Group class was tested to ensure that adding friends into groups was fully functional as well as changing group names. The Movie class was tested to see that the equals method worked well. This completes the unit testing. 
+
+All of our data was stored in mongoDB, so we decided it was a great idea to test the functionality of the database with our data. In order to do that we needed an instance of mongoDB in order to run these tests automatically every time a pull request was submitted. So we managed to create integration tests. We saved Sessions, Movies, Ratings, Users, Groups, respectively in their own collections and we performed the following tests. We made sure to test the loading, saving, updating, and dropping of these data sets. The way we implemented this was by creating an AbstractModel that would handle those methods no matter what kind of data was being manipulated. 
```

Comment:
Instead of "we managed to create integration tests" you could say "we added an embedded MongoDB instance to run our tests against"